modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -550,6 +550,10 @@ def groupby_agg(self, by, axis, agg, groupby_args, **kwargs):
                 + f"met '{type(by_frame._op).__name__}'."
             )
 
+        if agg in ("head", "tail"):
+            n = kwargs["agg_kwargs"]["n"]
+            return self._groupby_head_tail(agg, n, groupby_cols.tolist())
```

Comment:
Why do this conversion?

```suggestion
            return self._groupby_head_tail(agg, n, groupby_cols)
```